containers/unit-tests/start

```diff
@@ -1,33 +1,199 @@
+        super().__init__(prefix='cockpit-source.')
+
+    def prepare(self, args):
+        if args.branch:
+            opts = ['-c', 'advice.detachedHead=false', '-b', args.branch]
+        else:
+            opts = []
+
+        git('clone', *opts, '.', self.name)
+
+        if not args.head and not args.branch:
+            if stash := git_output('stash', 'create'):
+                git('-C', self.name, 'fetch', '--quiet', '--no-write-fetch-head', 'origin', stash)
+                git('-C', self.name, 'stash', 'apply', stash)
+
+        if not args.no_node_modules:
+            run([f'{self.name}/tools/node-modules', 'checkout'], check=True)
+
+            if args.wait:
+                run([f'{self.name}/tools/webpack-jumpstart', '--wait'], check=True)
+            elif not args.no_jumpstart:
+                run([f'{self.name}/tools/webpack-jumpstart', '--partial'], check=True)
+
+
+class ResultsDirectory(PodmanTemporaryDirectory):
+    def __init__(self):
+        super().__init__(prefix='cockpit-results.')
+
+    def copy_out(self, destination):
+        podman('unshare', 'cp', '-rT', self.name, destination)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--verbose', '-v', action='store_true', help='Show commands when running them')
+    parser.add_argument('--results', metavar='DIRECTORY', help="Copy container /results to the given host directory")
+
+    group = parser.add_argument_group(title='Container options')
+    group.add_argument('--image', default='ghcr.io/cockpit-project/unit-tests', help='Container image to use')
+    group.add_argument('--image-tag', default='latest', help='Container image tag to use')
+    group.add_argument('--env', metavar='NAME=VAL', action='append', default=[],
+                       help='Set an environment variable in the container')
+    group.add_argument('--network', action="store_true",
+                       help="Enable network in the container (default: disabled)")
+    group.add_argument('--interactive', '-i', action="store_true",
+                       help="Interactive mode (implied by no command or script)")
+    group.add_argument('--tty', '-t', action="store_true",
+                       help="Allocate a pseudoterminal (implied by no command or script)")
+    group.add_argument('--user', help="Pass through the --user flag to podman")
+    group.add_argument('--entrypoint', metavar='CMD', help="Provide the --entrypoint flag to podman")
+    group.add_argument('--workdir', help="Provide the --workdir flag to podman")
+
+    group = parser.add_argument_group(title='What to build').add_mutually_exclusive_group()
+    group.add_argument('--head', action='store_true', help='Build the HEAD commit')
+    group.add_argument('-b', dest='branch', metavar='NAME', help='Build the named branch or tag')
+    group.add_argument('--work-tree', action='store_true',
+                       help='Build the HEAD commit, plus changes on the filesystem (default)')
```

Comment:
This also feels like confusing duplication and overhead glue to me. There's a zillion way of testing a git tree, with local changes, local or remote branches or whatnot. Every other `make check` in the world just runs the current checkout.

Reply:
The problem is that we can't just use the current checkout because we can't get it inside of the container without relabelling it.  We could definitely trim down some of the options here, though.